ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/rally/DecryptPayloads.java

```diff
@@ -0,0 +1,91 @@
+package com.mozilla.telemetry.decoder.rally;
+
+import com.mozilla.telemetry.ingestion.core.Constant.Attribute;
+import com.mozilla.telemetry.transforms.PubsubConstraints;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.transforms.Flatten;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.Partition;
+import org.apache.beam.sdk.transforms.WithFailures;
+import org.apache.beam.sdk.transforms.WithFailures.Result;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionList;
+
+public class DecryptPayloads extends
+    PTransform<PCollection<PubsubMessage>, Result<PCollection<PubsubMessage>, PubsubMessage>> {
+
+  private final ValueProvider<String> metadataLocation;
+  private final ValueProvider<String> schemasLocation;
+  private final ValueProvider<Boolean> kmsEnabled;
+  private final ValueProvider<Boolean> decompressPayload;
+
+  public static DecryptPayloads of(ValueProvider<String> metadataLocation,
+      ValueProvider<String> schemasLocation, ValueProvider<Boolean> kmsEnabled,
+      ValueProvider<Boolean> decompressPayload) {
+    return new DecryptPayloads(metadataLocation, schemasLocation, kmsEnabled, decompressPayload);
+  }
+
+  private DecryptPayloads(ValueProvider<String> metadataLocation,
+      ValueProvider<String> schemasLocation, ValueProvider<Boolean> kmsEnabled,
+      ValueProvider<Boolean> decompressPayload) {
+    this.metadataLocation = metadataLocation;
+    this.schemasLocation = schemasLocation;
+    this.kmsEnabled = kmsEnabled;
+    this.decompressPayload = decompressPayload;
+  }
+
+  @Override
+  public Result<PCollection<PubsubMessage>, PubsubMessage> expand(
+      PCollection<PubsubMessage> messages) {
+
+    final List<PCollection<PubsubMessage>> failureCollections = new ArrayList<>();
+
+    PCollectionList<PubsubMessage> partitioned = messages //
+        .apply("PartitionRallyAndPioneer", Partition.of(2, RallyPartitioner.of()));
```

Comment:
```suggestion
    PCollectionList<PubsubMessage> partitioned = messages //
        .apply("PartitionRallyAndPioneer", Partition.of(2,
            (PubsubMessage message, int numPartitions) -> isPioneerPing(message) ? 1 : 0));
```

I think this is more clear as a lambda, especially considering that the `1` and `0` here are referenced again on the following lines, so keeping those as close together in the code as possible will aid understanding.

Reply:
Very nice, no need for the extra typing information.